src/main/java/org/reactome/release/cosmicupdate/Main.java

```diff
@@ -0,0 +1,337 @@
+	
+	@Parameter(names = {"-c"}, description = "The path to the configuration file. Default is src/main/resources/config.properties")
+	private String configPath = "src/main/resources/config.properties";
+	
+//	@Parameter(names = {"-a"}, description = "The path to the authorization file, containing usernames and passwords. Default is src/main/resources/auth.properties")
+//	private static String authPath = "src/main/resources/auth.properties";
+	
+	private static String COSMICMutantExport;
+	private static String COSMICFusionExport;
+	private static String COSMICMutationTracking;
+
+	private static String COSMICMutantExportURL;
+	private static String COSMICFusionExportURL;
+	private static String COSMICMutationTrackingURL;
+	
+	private static String COSMICUsername;
+	private static String COSMICPassword;
+	
+	private static final Logger logger = LogManager.getLogger();
+	private static long personId;
+
+	public static void main(String... args)
+	{
+		try
+		{
+			Main cosmicUpdateStep = new Main();
+			JCommander.newBuilder().addObject(cosmicUpdateStep).build().parse(args);
+			Properties configProps = cosmicUpdateStep.getConfig();
+			cosmicUpdateStep.executeStep(configProps);
+		}
+		catch (SQLException e)
+		{
+			logger.error("Error with SQL Connection", e);
+			e.printStackTrace();
+		} catch (Exception e) {
+			logger.error("Exception caught", e);
+		}
+		logger.info("All done.");
+		
+	}
+
+	private Properties getConfig()
+	{
+		boolean noConfigSpecified = this.configPath != null && this.configPath.trim().isEmpty();
+//		boolean noAuthConfigSpecified = Main.authPath != null && !Main.authPath.trim().isEmpty();
+		if (!noConfigSpecified /* && !noAuthConfigSpecified */)
+		{
+			try(Reader configReader = new FileReader(this.configPath);
+			/* Reader authReader = new FileReader(Main.authPath) */)
+			{
+				Properties configProps = new Properties();
+				configProps.load(configReader);
+				
+//				Properties authProps = new Properties();
+//				authProps.load(authReader);
+
+				// These default to files in the same directory where the program is running.
+				Main.COSMICMutantExport = configProps.getProperty("pathToMutantExportFile", "./CosmicMutantExport.tsv");
+				Main.COSMICFusionExport = configProps.getProperty("pathToFusionExportFile", "./CosmicFusionExport.tsv");
+				Main.COSMICMutationTracking = configProps.getProperty("pathToMutationTrackingFile", "./CosmicMutationTracking.tsv");
+				// These have NO default.
+				Main.COSMICUsername = configProps.getProperty("cosmic.user");
+				Main.COSMICPassword = configProps.getProperty("cosmic.password");
+				Main.personId = Long.parseLong(configProps.getProperty("personId"));
+				// These must be set in the properties file.
+				Main.COSMICMutantExportURL = configProps.getProperty("urlToMutantExportFile");
+				Main.COSMICFusionExportURL = configProps.getProperty("urlToFusionExportFile");
+				Main.COSMICMutationTrackingURL = configProps.getProperty("urlToMutationTrackingFile");
```

Comment:
Should these be checked to see if they received values and an exception thrown if not?

Reply:
Re-working exception handling, in a larger scope...